src/celeritas/grid/XsCalculator.hh

```diff
@@ -83,21 +86,33 @@ XsCalculator::XsCalculator(const XsGridData& grid, const Values& values)
  *
  * If needed, we can add a "log(energy/MeV)" accessor if we constantly reuse
  * that value and don't want to repeat the `std::log` operation.
+ *
+ * \todo Do we ever need to extrapolate cross sections beyond the energy
+ * bounds? (And is it OK if those endpoints are a bit fuzzy because of the
+ * logarithms?)
  */
 CELER_FUNCTION real_type XsCalculator::operator()(Energy energy) const
 {
     const UniformGrid loge_grid(data_.log_energy);
     const real_type   loge = std::log(energy.value());
 
-    // Snap out-of-bounds values to closest grid points
+    // Cross section is zero outside of defined grid
```

Comment:
Now I'm not totally convinced this is the right behavior. Since Geant4 extrapolates rather than zeros outside the grid, maybe as long as we are importing our physics tables from Geant4 we should do the same thing? Looking at how they build the lambda tables, they use a "start from null" flag to indicate whether the first cross section value on the table should be zero (presumably so that extrapolating will produce the correct result below the threshold for that process) -- however, it's not true for all processes. Also, should the cross section really be zero above the energy grid? It's a little awkward too to change the behavior of the eloss calculator so that below the grid the dE/dx is zero rather than the first value and then hack in a way to set it back to the first value.

Reply:
ok, maybe @stognini we also should add that flag to the export data. Is it exposed as an accessor?

Lol yes it is awkward. Let me split off the changes that *don't* require touching the model/cross section extrapolation, and then I can improve what we're doign here

Reply:
> Is it exposed as an accessor?

Hmm no. It also only apples if the minimum primary energy for the process is greater than the minimum kinetic energy of the tables. But I guess whatever the first xs value is is what should be used below the energy grid.

Reply:
As @amandalund  said, the `startFromNull` flags in both `G4VEmProcess::BuildLambdaTable()` and `G4VEenergyLossProcess::BuildLambdaTable()` are local variables. To make things worse, we can't use the `MinPrimaryEnergy(...)` to replicate the criterion just mentioned by @amandalund at import time, because these are a protected members of each class. So it is not (currently) obvious how we could get these flag values out, but I'll dig more to try and find a workaround without any changes to G4.

Also, not directly related to this topic, but might be relevant to the conversation: G4 does a similar thing to micro-xs. See comment here:

https://github.com/celeritas-project/celeritas/blob/master/src/celeritas/ext/detail/ImportProcessConverter.cc#L618-L641

Reply:
@stognini If I'm understanding you and your comment correctly, do we need to have flags for whether to set off-the-end values to zero? If "Geant4 simply uses the next/previous bin value when the vector's front/back are zero" when inside the energy range, and if it uses the actual first and last value outside the energy range, then the current implementation in celeritas will give a different (nonzero) value when accessing out-of-range energies.

Reply:
If we don't do exactly the same as Geant4, then it is useful to know about any forced rewrite in the physics vectors: Either a zero added at the front of a macro lambda table, or repeated non-zero values at the front/back of micro-xs tables.

Did we reach a consensus on how we want to deal with process limits, as discussed in #356 ?